client.c

```diff
@@ -29,6 +29,10 @@ int main()
     for (int i = 0; i <= offset; i++) {
         lseek(fd, i, SEEK_SET);
         sz = read(fd, buf, 1);
+        if (sz < 0) {
```

Comment:
This should appear in another pull request. Let's stick to the proposed subject for necessary changes.